dashboard/app/models/user.rb

```diff
@@ -731,7 +731,15 @@ def advertised_scripts
   def in_progress_and_completed_scripts
     backfill_user_scripts if needs_to_backfill_user_scripts?
 
-    user_scripts.compact
+    user_scripts.compact.reject do |user_script|
+      begin
+        user_script.script.nil?
```

Comment:
My fault, I assumed a DB query was being generated based on the PR description (that mentions additional DB call). Given that `UserScript.script` uses the cache, LGTM!